Architecture refactor

FrankUrbach opened the pull request:
The reason for this pull request is a ground breaking change in the structure of the java client. The Grakn team asks us to follow their structure if it's possible. 

- For more clarity the protocol proto files are also changed.  
- Changed the base of the client from the python to the java client
- take over the sources in the Julia files as comments. That should ease to follow which logic should be implemented there.

FrankUrbach committed:
First commit to align Julia client with the Java client
after reworking the structure of the Java client

FrankUrbach committed:
protobufs enabled

FrankUrbach committed:
First base GraknClient.jl

FrankUrbach committed:
added generated file grakn_pb.jl is missing in the
right shape

FrankUrbach committed:
correct grakn_pb.jl and grakn.jl in generated
files included

FrankUrbach committed:
changed the docs back to the original Julia doc

FrankUrbach committed:
Next step towards automatic translation of the
structure of the grakn java client.

FrankUrbach committed:
take the contents of .gitignore from the
main branch

FrankUrbach committed:
remove .DS_Store and Manifest.toml from the
repository and changed the .gitignore file according
the main path

FrankUrbach committed:
Next step towards changing structure according
the java client implementation.

FrankUrbach committed:
removed a little glitch within grakn.jl in the
generated folder

FrankUrbach committed:
changed location of the test folder

FrankUrbach committed:
final translation of the java structure to Juila

FrankUrbach committed:
commit all changed files for translation

FrankUrbach committed:
final translation of the client java structur for
the julia client

FrankUrbach committed:
Final translation of the java client structure
for the julia client.

FrankUrbach committed:
deleted Revise entry from Project.toml

FrankUrbach committed:
implementation fo the Errormessage is done

FrankUrbach committed:
Implemented Internal error section

mkschulze reviewed (approved):
Hi Frank, it looks very good, just please recheck if all .Ds_Store files are removed and manifest files are not checked in, especially in dev,test and docs. And that the docs are basically untouched. Thx

tk3369 commented on Project.toml:
```diff
@@ -4,12 +4,13 @@ authors = ["Humans of Julia"]
 
 [deps]
+Compat = "34da2185-b29b-5c13-b0c7-acf172513d20"
+DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
```
I don't think these are used?

FrankUrbach replied:
You are right. As of today, we are not yet using either package. I deleted them

tk3369 commented on src/common/exception/ErrorMessage.jl:
```diff
@@ -0,0 +1,220 @@
+# The Symbol left side is assigned to the struct which will be build bellow
+# The right side of the Pair is assigned to the abstract type representing the section of error
+const error_structs = (
+    :Client_CLIENT_CLOSED=>:AbstractClientError,
```
Sample style suggestions. Use UPPER_CASE for constants and put spaces around operators.

```suggestion
const ERROR_STRUCTS = (
    :CLIENT_CLIENT_CLOSED => :AbstractClientError,
```

tk3369 commented on src/common/exception/ErrorMessage.jl:
```diff
@@ -0,0 +1,220 @@
+# All structs represent a unique number in a section of errors and the description. Both of them are 
+# assigned on the right side pair
+const error_messages = Dict([
+        Client_CLIENT_CLOSED=>( 1, "The client has been closed and no further operation is allowed."),
```
Please unindent 4 spaces and similar style suggestions below.
```suggestion
        CLIENT_CLIENT_CLOSED => (1, "The client has been closed and no further operation is allowed."),
```

tk3369 commented on src/common/exception/ErrorMessage.jl:
```diff
@@ -0,0 +1,220 @@
+
+# Build the error massage according the section of errors represented by the 
+# abstract type of each section of errors
+function _buidl_error_messages(class::Type{T}) where {T<:AbstractClientError}
```
Typo.. should be `build` not `buidl`. Same comments for functions below.

tk3369 replied:
I also don't understand what it's building... It's returning a single instance of the class but it's not passing any variable arguments for the string template. Perhaps this is still WIP?

It's nice start though! Let me think a little more about this design and come back with more suggestions later.

FrankUrbach replied:
Maybe it will be more clear if we implement the next file GraknClientException.jl. The usage of this will be more clear there.

FrankUrbach committed:
Update src/common/exception/ErrorMessage.jl

Co-authored-by: Tom Kwong <tk3369@gmail.com>

FrankUrbach committed:
Update src/common/exception/ErrorMessage.jl

Co-authored-by: Tom Kwong <tk3369@gmail.com>

FrankUrbach committed:
removed some unnessecary packages from the
Project.toml, corrected the typo in _build_error_messages (Thanks Tom)
and changed the gitignore file to prevent the unwanted .DS_Store files.

FrankUrbach committed:
Merge branch 'architecture-refactor' of https://github.com/Humans-of-Julia/GraknClient.jl into architecture-refactor
